pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -87,10 +86,23 @@ func (az *Cloud) ExternalID(name types.NodeName) (string, error) {
 	return az.InstanceID(name)
 }
 
-// InstanceExistsByProviderID returns true if the instance with the given provider id still exists and is running.
+// InstanceExistsByProviderID returns true if the instance with the given provider id still exists.
 // If false is returned with no error, the instance will be immediately deleted by the cloud controller manager.
 func (az *Cloud) InstanceExistsByProviderID(providerID string) (bool, error) {
-	return false, errors.New("unimplemented")
+	name, err := splitProviderID(providerID)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = az.InstanceID(name)
+	if err != nil {
+		if err == cloudprovider.InstanceNotFound {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
```

Comment:
Is it possible to add tests for this function?

Reply:
It is too simple and there is no good test case for this function.